File: Strivers A2Z DSA Sheet/Patterns Printing/Pattern02.cpp
#include <iostream>
using namespace std;

class Pattern02
{
private:
    int iSize;

public:
    Pattern02(int iNo)
    {
        iSize = iNo;
    }

    void Pattern02NxN()
    {
        for (int i = 0; i <= iSize; i++)
        {
            for (int j = 0; j <= i; j++)
            {
                cout << "*\t";
            }
            cout << endl;
        }
    }
};

int main()
{
    int iValue = 0;
    cout << "Enter the Size : ";
    cin >> iValue;
    Pattern02 *pobj = new Pattern02(iValue);
    pobj->Pattern02NxN();
    delete (pobj);

    return 0;
}

File: Strivers A2Z DSA Sheet/Patterns Printing/Pattern02.java
import java.util.*;

class Pattern02Nxn
{
    private int iSize = 0;

    public Pattern02Nxn(int iNo)
    {
        this.iSize = iNo;
    }

    public void Pattern02Print()
    {
        for(int i = 0; i< iSize;i++)
        { 
            for (int j = 0;j<= i;j++)
            {
                System.out.print("*\t");
            }System.out.println();
        }
    }
}

class Pattern02 {
    public static void main(String Args[]) {
        int iValue = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter the iValue : ");
         iValue = sobj.nextInt();

        Pattern02Nxn pobj = new Pattern02Nxn(iValue);
        pobj.Pattern02Print();
        sobj .close();

    }

}
